### ISSUE TITLE
图文匹配任务 输入单图和单条文本 判断这个pair是否匹配 应该如何做

### ISSUE THREAD

**Issue by kiddliao** [OFA-Sys/Chinese-CLIP#189]
如果输入单图和多个文本 判断图片的分类 我使用的代码如下
import torch
from PIL import Image

import cn_clip.clip as clip
from cn_clip.clip import load_from_name, available_models
print("Available models:", available_models())
# Available models: ['ViT-B-16', 'ViT-L-14', 'ViT-L-14-336', 'ViT-H-14', 'RN50']

device = "cuda" if torch.cuda.is_available() else "cpu"
model, preprocess = load_from_name(r"RN50", device=device, download_root='./')
model.eval()
image = preprocess(Image.open("1.jpg")).unsqueeze(0).to(device)
text = clip.tokenize(["短袖", "手镯", "鞋", "1", "运动鞋"]).to(device)

with torch.no_grad():
    image_features = model.encode_image(image)
    text_features = model.encode_text(text)
    # 对特征进行归一化，请使用归一化后的图文特征用于下游任务
    image_features /= image_features.norm(dim=-1, keepdim=True)
    text_features /= text_features.norm(dim=-1, keepdim=True)

    logit_scale = model.logit_scale.exp()
    logits_per_image = logit_scale * image_features @ text_features.t()
    logits_per_text = logits_per_image.t()
    # print(logits_per_image.tolist())
    probs = logits_per_image.softmax(dim=-1).cpu().numpy()

print("Label probs:", probs)  # [[1.268734e-03 5.436878e-02 6.795761e-04 9.436829e-01]]

如果只希望判断输入的单张图和单个文本是否匹配 则不能使用softmax
我打印出logits_per_image的值 大概是40左右 如何有效地转换为0-1的概率呢

**Comment by 1010936691** [comment]
我用这个方式获取一个文本和一个图片的相似度，总是得到1.0

**Comment by 1010936691** [comment]
> 我用这个方式获取一个文本和一个图片的相似度，总是得到1.0

这是我的代码（其实是git本身的代码，我应该没有去动过这里model.py）
    def get_similarity(self, image, text):
        image_features = self.encode_image(image)
        text_features = self.encode_text(text)

        # normalized features
        image_features = image_features / image_features.norm(dim=1, keepdim=True)
        text_features = text_features / text_features.norm(dim=1, keepdim=True)

        # cosine similarity as logits
        logit_scale = self.logit_scale.exp()
        logits_per_image = logit_scale * image_features @ text_features.t()
        logits_per_text = logits_per_image.t()

        # shape = [global_batch_size, global_batch_size]
        return logits_per_image, logits_per_text
